python/artexplainer.Dockerfile

```diff
@@ -33,18 +37,18 @@ RUN cd artexplainer && poetry install --no-interaction --no-cache
 
 FROM ${BASE_IMAGE} as prod
 
-COPY third_party third_party
-
-# Activate virtual env
-ARG VENV_PATH
-ENV VIRTUAL_ENV=${VENV_PATH}
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ARG WORK_DIR
+WORKDIR ${WORK_DIR}
 
 RUN useradd kserve -m -u 1000 -d /home/kserve
+USER 1000
 
-COPY --from=builder --chown=kserve:kserve $VIRTUAL_ENV $VIRTUAL_ENV
-COPY --from=builder kserve kserve
-COPY --from=builder artexplainer artexplainer
+# To activate virtual env we have to set VIRTUAL_ENV environment variable
+ARG VENV_PATH
+ENV VIRTUAL_ENV=${WORK_DIR}/${VENV_PATH}
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
+
+COPY third_party third_party
+COPY --from=builder --chown=kserve:0 ${WORK_DIR} .
 
```

Comment:
There seem to be a lot of redundancy among these Dockerfiles. Should we provide a base-python-Dockerfile that has all the common layers?

Reply:
There is already a PR addressing this issue https://github.com/kserve/kserve/pull/3114. But this approach has one down side. If developers are using `docker-container` driver for building images then the base image has to be pushed to a local registry or remote registry to build the model server images. You can find more info on this in the [PR comment](https://github.com/kserve/kserve/pull/3114#discussion_r1337069115).